src/inmanta/references.py

```diff
@@ -313,13 +339,14 @@ def is_dataclass_reference(cls) -> bool:
         return dataclasses.is_dataclass(cls.get_reference_type())
 
     @abc.abstractmethod
-    def resolve(self) -> T:
+    def resolve(self, ctx: "handler.LoggerABC") -> T:
```

Comment:
Fairly minor, but shouldn't be name this argument `logger` if it is just a logger?  (I know the actual object is a context when called in the agent, but it won't be it when called in the compiler, and I feel like calling ctx might then cause confusion)

Reply:
Did you push?